src/plugins/intel_cpu/src/nodes/eye.cpp

```diff
@@ -144,7 +144,7 @@ void Eye::executeSpecified() {
         parallel_nt(0, [&](const size_t ithr, const size_t nthr) {
             size_t start = 0, end = 0;
             splitter(batchVolume, nthr, ithr, start, end);
-            memset(dst + start * spatialCount, 0, (end - start) * spatialSize);
+            memset(reinterpret_cast<void*>(dst + start * spatialCount), 0, (end - start) * spatialSize);
```

Comment:
for what type do have this warning?

Reply:
Only **template** and **auto** will trigger warning, I aslo search memset in other files, I found that if input varable is specific, for example int or float, there is no warning.

Reply:
I understand that float and int lead to no warnings, but I saw the warnings on bfloat types, which should not be used in CPU plugin. 
So, instead of simple suppression we need to understand whether the template specialization with specific type is really needed.

Reply:
Hi @ilya-lavrenov 
No, bfloat16_t is used  in CPU plugin. Actually SPR's fp32 model will be ran at default bf16 precision with AMX instruct for  some computive Nodes(Conv, FC, MatMul,...).
Currently, bfloat16_t is not native type of compiler, it is implemented at: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/utils/bfloat16.hpp#L21

About why it have warning when using memset, it is related to a cpp concept trivial(https://en.cppreference.com/w/cpp/types/is_trivial,  https://en.cppreference.com/w/cpp/named_req/TrivialType, https://en.cppreference.com/w/cpp/language/classes#Trivial_class).

Unfortunately, our current bfloat16_t's implementation is not trivial.
So here are 2 solutions to fix it.
1: My current implement, using reinterpret_cast to convert forcely.
2: Update class bfloat16_t's implement, below is my update, it will be changed to trivial.
```
-    constexpr bfloat16_t()
-        : m_value{0}
-    {
-    }
+    bfloat16_t() = default;
```

What would you like? or have other ideas?

Reply:
BTW, it also related with compiler version, low version compiler(4.9, 5.2, 7.5) don't tip any warnings, but high version compiler (12.1 ) will have warnings.

Reply:
I'm not sure that CPU needs template instantiations with bfloat16, because my expectation they will be quite slow. 
CPU plugin usually provides optimized kernels, so do you consider Eye::executeSpecified<bfloat16_t> as optimized?

Reply:
CPU bfloat16_t itself is wrapper of uint16_t, I don't why it's instantiation will be slow. 
It is similar to oneDNN bfloat16_t(I guess it copy from oneDNN). 
@dmitry-gorokhov @usstq what do you think?

Reply:
Hi @ilya-lavrenov , I understand your concern.
1: CPU plugin itself need bf16 value type;
For example:
[src/plugins/intel_cpu/src/nodes/cum_sum.cpp:103](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/cum_sum.cpp#L103)
[src/plugins/intel_cpu/src/nodes/eye.cpp:84](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/eye.cpp#L84)
...
2: Some node instance bf16, I think it is base implement, almost case will not call base implement.
For example:
[src/plugins/intel_cpu/src/nodes/mha.cpp:1215](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mha.cpp#L1215)
convertTransposeKernel will not be nullptr, 
Please check: 
[src/plugins/intel_cpu/src/nodes/mha.cpp:1125](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mha.cpp#L1125)